inference-engine/src/mkldnn_plugin/nodes/mkldnn_extract_image_patches_node.cpp

```diff
@@ -419,90 +387,215 @@ void MKLDNNExtractImagePatchesNode::initSupportedPrimitiveDescriptors() {
 }
 
 void MKLDNNExtractImagePatchesNode::execute(mkldnn::stream strm) {
-    const char *src_data = reinterpret_cast<const char *>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr());
-    char *dst_data = reinterpret_cast<char *>(getChildEdgesAtPort(0)[0]->getMemoryPtr()->GetPtr());
-    const size_t dtype_size = getOriginalInputPrecisionAtPort(0).size();
-
-    const auto& inDims = getParentEdgeAt(0)->getMemory().getStaticDims();
-    const size_t IC = inDims[1];
-    const size_t IH = inDims[2];
-    const size_t IW = inDims[3];
-
-    const auto& outDims = getChildEdgesAtPort(0)[0]->getMemory().getStaticDims();
-    const size_t OB = outDims[0];
-    const size_t OH = outDims[2];
-    const size_t OW = outDims[3];
-
-    const size_t KH = _ksizes[0], KW = _ksizes[1];
-    const size_t SH = _strides[0], SW = _strides[1];
-    const size_t RH = _rates[0], RW = _rates[1];
-    const size_t PT = _pad_top, PL = _pad_left;
-
-    const std::vector<size_t> istrides = getParentEdgeAt(0)->getMemory().GetDescWithType<BlockedMemoryDesc>()->getStrides();
-    const std::vector<size_t> ostrides = getChildEdgesAtPort(0)[0]->getMemory().GetDescWithType<BlockedMemoryDesc>()->getStrides();
-    const std::vector<size_t> ostrides_partial = {ostrides[0], KW * IC * ostrides[1], IC * ostrides[1], ostrides[1]};
-
-    if (extract_image_patches_kernel) {
-        parallel_for4d(OB, KH, KW, IC, [&](const size_t ob, const size_t kh, const size_t kw, const size_t ic) {
-            const int64_t ih_start = kh * RH - PT;
-            const int64_t iw_start = kw * RW - PL;
-            const size_t ih_lpad = ih_start >= 0 ? 0 : std::ceil(-1.f * ih_start / SH);
-            const size_t iw_lpad = iw_start >= 0 ? 0 : std::ceil(-1.f * iw_start / SW);
-            const size_t ih_hpad = std::ceil((IH - 1.f * ih_start) / SH) > OH ? OH : std::ceil((IH - 1.f * ih_start) / SH);
-            const size_t iw_hpad = std::ceil((IW - 1.f * iw_start) / SW) > OW ? OW : std::ceil((IW - 1.f * iw_start) / SW);
-
-            size_t dst_offset = ob * ostrides_partial[0] + kh * ostrides_partial[1] + kw * ostrides_partial[2] + ic * ostrides_partial[3];
-            size_t src_offset = ob * istrides[0] + ic * istrides[1] + ih_start * istrides[2] + iw_start + ih_lpad * SH * IW;
-
-            auto args = jit_extract_image_patches_args();
-            args.src = src_data + src_offset * dtype_size;
-            args.dst = dst_data + dst_offset * dtype_size;
-            args.h_lo_pad = ih_lpad;
-            args.h_hi_pad = ih_hpad;
-            args.w_lo_pad = iw_lpad;
-            args.w_hi_pad = iw_hpad;
-            (*extract_image_patches_kernel)(&args);
-        });
+    if (execPtr) {
+        auto src = getParentEdgeAt(0)->getMemoryPtr()->GetPtr();
+        auto dst = getChildEdgesAtPort(0)[0]->getMemoryPtr()->GetPtr();
+        const auto& inStrides = getParentEdgeAt(0)->getMemory().GetDescWithType<BlockedMemoryDesc>()->getStrides();
```

Comment:
`GetDescWithType` return pointer to object which can be destroyed
so better to copy

Reply:
Fixed, thanks